File: read_excel.py
import openpyxl
import sys

try:
    sys.stdout.reconfigure(encoding='utf-8')
except:
    pass

# Load Excel file
wb = openpyxl.load_workbook(r'C:\Code\Sup_AI_Edu\assets\Giảng viên\danh_sach_giang_vien_day_du_704_dong.xlsx')
ws = wb.active

print(f'📊 Sheet name: {ws.title}')
print(f'📈 Total rows: {ws.max_row}')
print(f'📋 Total columns: {ws.max_column}')
print()
print('🔍 Column headers:')
for col in range(1, ws.max_column + 1):
    header = ws.cell(1, col).value
    print(f'  Col {col}: {header}')

print()
print('📝 First 3 rows of data:')
for row in range(2, min(5, ws.max_row + 1)):
    row_data = []
    for col in range(1, ws.max_column + 1):
        cell_value = ws.cell(row, col).value
        row_data.append(str(cell_value)[:25] if cell_value else 'N/A')
    print(f'  Row {row}: {" | ".join(row_data)}')

File: read_excel.js
const XLSX = require('xlsx');
const path = require('path');

// Đường dẫn tệp Excel giảng viên
const excelPath = path.join(__dirname, 'assets', 'Giảng viên', 'danh_sach_giang_vien_day_du_704_dong.xlsx');

try {
    const workbook = XLSX.readFile(excelPath);
    const sheetName = workbook.SheetNames[0];
    const sheet = workbook.Sheets[sheetName];
    
    console.log(`📊 Sheet name: ${sheetName}`);
    
    const range = XLSX.utils.decode_range(sheet['!ref']);
    console.log(`📈 Total rows: ${range.e.r + 1}`);
    console.log(`📋 Total columns: ${range.e.c + 1}`);
    console.log();
    
    console.log('🔍 Column headers:');
    for (let col = range.s.c; col <= range.e.c; col++) {
        const cellAddress = XLSX.utils.encode_cell({ r: range.s.r, c: col });
        const cell = sheet[cellAddress];
        console.log(`  Col ${col + 1}: ${cell ? cell.v : 'N/A'}`);
    }
    
    console.log();
    console.log('📝 First 3 rows of data:');
    for (let r = range.s.r + 1; r <= Math.min(range.s.r + 3, range.e.r); r++) {
        const rowData = [];
        for (let col = range.s.c; col <= range.e.c; col++) {
            const cellAddress = XLSX.utils.encode_cell({ r: r, c: col });
            const cell = sheet[cellAddress];
            const cellVal = cell ? cell.v : 'N/A';
            rowData.push(String(cellVal).substring(0, 25));
        }
        console.log(`  Row ${r + 1}: ${rowData.join(' | ')}`);
    }
} catch (e) {
    console.error(`Error reading ${excelPath}:`, e.message);
}
